commandapi-core/src/main/java/dev/jorel/commandapi/config/ConfigGenerator.java

```diff
@@ -0,0 +1,82 @@
+package dev.jorel.commandapi.config;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@ApiStatus.Internal
+public class ConfigGenerator {
+
+	private final DefaultedConfig defaultedConfig;
+
+	private ConfigGenerator(DefaultedConfig defaultedConfig) {
+		this.defaultedConfig = defaultedConfig;
+	}
+
+	public static ConfigGenerator createNew(DefaultedConfig defaultedConfig) {
+		return new ConfigGenerator(defaultedConfig);
+	}
+
+	public <T, C extends DefaultedConfig> void populateDefaultConfig(ConfigurationAdapter<T, C> adapter) {
+		for (Map.Entry<String, CommentedConfigOption<?>> commentedConfigOption : defaultedConfig.getAllOptions().entrySet()) {
+			adapter.setValue(commentedConfigOption.getKey(), commentedConfigOption.getValue().option());
+			adapter.setComment(commentedConfigOption.getKey(), commentedConfigOption.getValue().comment());
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	public <T, C extends DefaultedConfig> ConfigurationAdapter<T, C> generateWithNewValues(ConfigurationAdapter<T, C> existingConfig) {
+		ConfigurationAdapter<T, C> updatedConfig = existingConfig.createNew();
+
+		boolean shouldRemoveValues = shouldRemoveOptions(existingConfig);
+
+		boolean wasConfigUpdated = false;
+		for (Map.Entry<String, CommentedConfigOption<?>> commentedConfigOption : defaultedConfig.getAllOptions().entrySet()) {
+			String path = commentedConfigOption.getKey();
+
+			// Update config option
+			if (existingConfig.contains(path)) {
+				updatedConfig.tryCreateSection(path, (C) defaultedConfig);
```

Comment:
Did you test that? Because there should be logic in place to handle comments updating.

Reply:
Yea, I did run the plugin with the config above and it didn't update. Note though that if anything else is changed other than that comment, it will correctly detect that the config needs to be updated, in which case it will fix the comment.

There is code to detect if a `CommentedConfigOption` has the wrong comment:

https://github.com/JorelAli/CommandAPI/blob/bfea661c7c694ddcac088f47353efa31f57d3b83/commandapi-core/src/main/java/dev/jorel/commandapi/config/ConfigGenerator.java#L54-L56

However, the same does not happen for `CommentedSection` comments, so if those are the only thing that changes `wasConfigUpdated` will still be `false`.